install-nimbus-cli.sh

```diff
@@ -0,0 +1,284 @@
+    # then use that.
+    my_dir=$(cd "$NIMBUS_INSTALL_DIR" && pwd)
+    echoProgress "Installing into $my_dir"
+  elif [[ "$(which "$BINARY_NAME")" != "" ]] ; then
+    existing=$(which "$BINARY_NAME")
+    # If the user is already using a version of the tool, then use that.
+    # Follow the softlinks to where the file actually sites.
+    existing=$(readlink -f "$existing")
+    my_dir=$(dirname "$existing")
+  else
+    # Otherwise, we need to find some place on the existing PATH.
+    my_path=$(echo "$PATH" | tr ':' '\n')
+
+    # Prefer, in order:
+    #   1. the XDG binary directory (~/.local/bin)
+    #   2. ~/bin
+    #   3. /usr/local/bin
+    if echo "${my_path}" | grep -q "$HOME/.local/bin" ; then
+        my_dir="$HOME/.local/bin"
+    elif echo "${my_path}" | grep -q "$HOME/bin" ; then
+        my_dir="$HOME/bin"
+    elif echo "${my_path}" | grep -q "/usr/local/bin" ; then
+        my_dir="/usr/local/bin"
+    else
+        # If we can't find even a /usr/local/bin (unlikely)
+        # then we'll create the directory, and add it to the
+        # PATH, as well as adding it to the shell's . rc file.
+        my_dir="$HOME/.local/bin"
+        echoProgress "Adding $my_dir to PATH in .${SHELL}rc"
+        echo "export PATH=$PATH:\"$my_dir\"" >> "${HOME}/.${SHELL}rc"
+        export PATH=$PATH:"$my_dir"
+    fi
+  fi
+
+  if [[ ! -d "$my_dir" ]] ; then
+    echoProgress "Creating $my_dir"
+    mkdir -p "$my_dir"
+  fi
+
+  # If the directory isn't writeable, then we'll need to use sudo.
+  if [[ ! -w "$my_dir" ]] ; then
+    USE_SUDO="true"
+  fi
+
+  NIMBUS_INSTALL_DIR="$my_dir"
+}
+
+# downloadFile downloads the latest binary package and also the checksum
+# for that binary.
+downloadFile() {
+  FILENAME="$BINARY_NAME-$SUFFIX.zip"
+  DOWNLOAD_URL="$TASKCLUSTER_HOST/api/index/v1/task/project.application-services.v2.${BINARY_NAME}.latest/artifacts/public/build/$FILENAME"
+  NIMBUS_TMP_ROOT=$(mktemp -dt "$BINARY_NAME-installer-XXXXXX")
+  NIMBUS_TMP_FILE="$NIMBUS_TMP_ROOT/$FILENAME"
+  echoProgress "Downloading $DOWNLOAD_URL"
+  if [ "${HAS_CURL}" == "true" ]; then
+    curl -SsL "$DOWNLOAD_URL" -o "$NIMBUS_TMP_FILE"
+  elif [ "${HAS_WGET}" == "true" ]; then
+    wget -q -O "$NIMBUS_TMP_FILE" "$DOWNLOAD_URL"
+  fi
+}
+
+cleanup() {
+  if [[ -d "${NIMBUS_TMP_ROOT:-}" ]]; then
+    echoProgress "Cleaning up installation directory"
+    rm -rf "$NIMBUS_TMP_ROOT"
+  fi
+}
+
+success() {
+  echoProgress "Success!"
+}
+
+# runs the given command as root (detects if we are root already)
+runAsRoot() {
+  if [ $EUID -ne 0 ] && [ "$USE_SUDO" = "true" ]; then
+    sudo "${@}"
+  else
+    "${@}"
+  fi
+}
+
+# installFile installs the nimbus-cli binary.
+installFile() {
+  NIMBUS_TMP="$NIMBUS_TMP_ROOT/unzipped"
+  mkdir -p "$NIMBUS_TMP"
+  unzip -qj "$NIMBUS_TMP_FILE" -d "$NIMBUS_TMP"
+  local binary
+  if [[ "$OS" == "windows" ]] ; then
+    binary="${BINARY_NAME}.exe"
```

Comment:
Does this installer support windows ;) If not we might want to return early with instructions?

Reply:
I've written it as though it does, but only via WSL: in which case, I think this works. If not, then the installer just won't work— it's a bash script!

Reply:
Oh I forgot to mention in a top-level comment but I did test the install on WSL and it worked!